scripts/reorganize_json.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3 -B
+
+'''
+For every JSON file identified in ARGV, ensure that it is located in the
+correct directory structure. If it isn't, move it to the correct directory.
+'''
+
+import sys
+import json
+from pathlib import Path
+
+from settings import output_file_path
+
+files = []
+if len(sys.argv) > 1:
+	for p in sys.argv[1:]:
+		path = Path(p)
+		if path.is_dir():
+			files += sorted(str(s) for s in path.rglob('*.json'))
+		else:
+			files.append(p)
+else:
+	files = sorted(Path(output_file_path).rglob('*.json'))
+
+for filename in files:
+	p = Path(filename)
+	with open(filename, 'r') as fh:
+		data = json.load(fh)
+		try:
+			uu = data['id']
+		except KeyError as e:
+			print(f'*** ERROR:{filename}: {e}')
+			continue
+		if not uu.startswith('urn:uuid:'):
+			raise Exception(f"file doesn't have a valid top-level UUID: {filename}")
```

Comment:
Is this necessary? Shouldn't it be checked previously in the process?  Or is this the place to do the checking?

Reply:
Not necessary. It is checked earlier. Mostly just a sanity check. It can be useful if this script is run outside of the normal pipeline process and the directory contains other JSON files (which has happened to me when the tag-URI to UUID JSON mapping file is in the same directory tree as the output files).

Reply:
Not to early-optimize but won't reading and parsing the file times 5 million files to check something that has already been checked take a not-insignificant amount of time?

What about a command line flag to use only in the pipeline to skip that check and instead just rely on the previous pass? That solves for the first issue. And the second issue ... if there's a JSON file that looks like a UUID and ends in .json that /isn't/ an output file ... something else is wrong.

Reply:
I can rework this to be based on just the filename. But compared to the time of the rest of the pipeline and even previous post-processing steps like the crom object merging, I think this probably *is* an insignificant amount of time. On my laptop, doing this renaming takes roughly a tenth of a millisecond per file. So that's roughly 10 minutes for 5 million files (and hopefully the AWS hardware is a bit better than my laptop).